packages/app/src/templates/reducers/template.reducer.ts

```diff
@@ -4,25 +4,27 @@
  */
 
 import { createReducer } from "@reduxjs/toolkit"
+import { DataSyncEvent } from "App/data-sync/constants"
+import { ReadAllIndexesAction } from "App/data-sync/reducers"
+import { TemplatesEvent, TemplateDeletingState } from "App/templates/constants"
 import {
   TemplateState,
+  DeleteTemplateAction,
   CreateTemplateRejectedAction,
   CreateTemplateFulfilledAction,
 } from "App/templates/reducers/template.interface"
-import { DataSyncEvent } from "App/data-sync/constants"
-import { ReadAllIndexesAction } from "App/data-sync/reducers"
 import {
   pendingAction,
   rejectedAction,
   fulfilledAction,
 } from "Renderer/store/helpers"
-import { TemplatesEvent } from "App/templates/constants"
 
 export const initialState: TemplateState = {
   data: [],
   loaded: false,
   loading: false,
   error: null,
+  deletingState: null,
```

Comment:
I remember that we discussed not using the extra states. You can check the status of the previous operation by comparing existing states.
Example:
```ts
Error: loaded === false && loading === false && error !== null
Progress: loaded === false && loading === true && error === null
Success: loaded === true && loading === false && error === null
```

Reply:
Yes, but I also need information that loaded, loading or error state is connected to deleting, not to add or editing template. So in this case I still need another state for example `deleting: boolean` 
Error: loaded === false && loading === false && error !== null && deleting 
Progress: loaded === false && loading === true && error === null && deleting 
Success: loaded === true && loading === false && error === null && deleting

Reply:
In each case responsible for deleting you can update the `loading/loaded` state too

Reply:
So, I removed deletingState, but added boolean deleting as in suggestion [02d3381](https://github.com/mudita/mudita-center/pull/770/commits/02d33817e550d87c036c35fb46efc745904b1e57)